chainerx_cc/chainerx/cuda/memory_pool.cc

```diff
@@ -107,39 +217,57 @@ void* MemoryPool::Malloc(size_t bytesize) {
                 throw OutOfMemoryError{bytesize};
             }
         }
+        chunk = std::make_unique<Chunk>(ptr, 0, allocation_size);
     }
 
+    CHAINERX_ASSERT(chunk != nullptr);
+    void* ptr = chunk->ptr();
```

Comment:
The variable name duplicates with the previous one. How about something different, e.g. `chunk_ptr`?